### ISSUE TITLE
opensipsctl  PgSQL

### ISSUE THREAD

**Issue by volga629** [OpenSIPS/opensips#1089]
Hello Everyone,
When use non standard port, opensipsctl stops working.

```
root@casbc00 ~> [~]#  opensipsctl dispatcher show
dispatcher gateways
pgsql_query: /usr/bin/psql  -A -q -t -P fieldsep='	' -h mydb.domain.com -p5235 -U opensips_prod opensips_prod01 -c 'select * FROM dispatcher ORDER BY setid; '
psql: could not connect to server: Connection refused
	Is the server running on host "mydb.domain.com" (10.28.145.200)
20) and accepting
	TCP/IP connections on port 5432?
```

**Comment by liviuchircu** [comment]
Are you sure you configured the `DBPORT` into `opensipsctlrc`? Here is how mine looks, and DB  creation works just fine:

```
DBENGINE=PGSQL

## database port (PostgreSQL=5432 default; MYSQL=3306 default)
DBPORT=5235

## database host
DBHOST=127.0.0.1

## database name (for ORACLE this is TNS name)
DBNAME=opensips

# database path used by dbtext, db_berkeley, or sqlite
# DB_PATH="/usr/local/etc/opensips/dbtext"

## database read/write user
DBRWUSER=liviu

## password for database read/write user
DBRWPW="mypassword"

## engine type for the MySQL/MariaDB tabels (default InnoDB)
# MYSQL_ENGINE="MyISAM"

## database super user (for ORACLE this is 'scheme-creator' user)
DBROOTUSER="liviu"
```

**Comment by liviuchircu** [comment]
Ugh... just tested the `dispatcher show` part, and it does seem to be broken ...

**Comment by volga629** [comment]
@liviuchircu 
Is this fix is in 2.3 ?

**Comment by liviuchircu** [comment]
yes (e9a1ed8618d)

**Comment by volga629** [comment]
I  tried fix and didn't helped

**Comment by liviuchircu** [comment]
Can you try to run the command manually, and see if you can get it to work? That's how I came up with the strange `-h` fix.

```
/usr/bin/psql  -A -q -t -P fieldsep='	' -h mydb.domain.com -p5235 -U opensips_prod opensips_prod01 -c 'select * FROM dispatcher ORDER BY setid; '
```

**Comment by volga629** [comment]
Yes, command come back with proper output 


````
root@casbc00 ~> [/etc/opensips]# /usr/bin/psql  -A -q -t -P fieldsep='' -hdbv01.networklab.prod -p5235 -U opensips_prod opensips_prod01 -c 'select * FROM dispatcher ORDER BY setid; '
71sip:10.18.130.26:51600fs://:@10.18.130.26:80210FS casip01
61sip:10.18.130.24:51600fs://:@10.18.130.24:80210FS casip00
````

**Comment by liviuchircu** [comment]
Are you sure it's not a `make install` problem? (i.e. you pulled the new code, but `/usr/local/opensips_proxy/lib/opensips/opensipsctl.pgsql` is unchanged, and has priority over `scripts/opensipsctl.pgsql`)

**Comment by volga629** [comment]
Yes, I am pretty sure rpm is delete source between the builds.

Here are link to our repo 

http://ftpsrv01.networklab.ca/fedora/25/RPMS/x86_64/

**Comment by liviuchircu** [comment]
Can you post the new output of `opensipsctl dispatcher show`?

**Comment by volga629** [comment]
root@casbc00 ~> [/etc/opensips]# opensipsctl dispatcher show
dispatcher gateways
pgsql_query: /usr/bin/psql  -A -q -t -P fieldsep='	' -hdbv01.networklab.prod -p 5235 -U opensips_prod opensips_prod01 -c 'select * FROM dispatcher ORDER BY setid; '
psql: could not connect to server: Connection refused
	Is the server running on host "dbv01.networklab.prod" (10.18.130.20) and accepting
	TCP/IP connections on port 5432?
root@casbc00 ~> [/etc/opensips]# date
Mon Apr 24 18:19:50 EDT 2017

**Comment by liviuchircu** [comment]
Can you run it by hand, and see what needs changing? `opensipsctl` is written in bash, so it's a good way of testing.

**Comment by liviuchircu** [comment]
Any updates here, @volga629 ? Did you successfully run the command-line query? If yes, how did you make it work?

**Comment by volga629** [comment]
The command run OK from shell. I am trying put -x to check where is calling pgsql and which script generate error 


````
/usr/bin/psql  -A -q -t -P fieldsep='' -hdbv01.networklab.prod -p5235 -U opensips_prod opensips_prod01 -c 'select * FROM dispatcher ORDER BY setid; '

71sip:10.18.130.26:51600fs://:CbnDpC0w7z@10.18.130.26:80210FS casip01
61sip:10.18.130.24:51600fs://:CbnDpC0w7z@10.18.130.24:80210FS casip00
````

**Comment by volga629** [comment]
I suspect problem around here are.
Trying find where set $DBCMD

```
1300 #
1301 ##### ------------------------------------------------ #####
1302 ### DISPATCHER management
1303 #
1304 dispatcher() {
1305         require_dbengine
1306         require_ctlengine
1307         case $1 in
1308                 show)
1309                         mecho "dispatcher gateways"
1310                         QUERY="select * FROM $DISPATCHER_TABLE ORDER BY $DISPATCHER_SETID_COLUMN; "
1311                         $DBCMD "$QUERY"
1312                         ;;

````

**Comment by volga629** [comment]
it like it try call /usr/sbin/pgsql twice

**Comment by volga629** [comment]
Here are working function

````
pgsql_query() {
        # if password not yet queried, query it now
        prompt_pw "PgSQL password for user '$DBRWUSER@$DBHOST'"
        mecho "pgsql_query: $PGSQL $2 -A -q -t -P fieldsep='    ' -h$DBHOST $PORT_OPT -U $DBRWUSER $DBNAME -c '$1'"
        if [ -z "$DBPORT" ]; then
                PGPASSWORD="$DBRWPW" $PGSQL $2 \
                            -A -q -t \
                            -P fieldsep="       " \
                            -h$DBHOST \
                            -U $DBRWUSER \
                             $DBNAME \
                            -c "$1"
        else
                             PGPASSWORD="$DBRWPW" $PGSQL $2 \
                            -A -q -t \
                            -P fieldsep="       " \
                            -h$DBHOST \
                            $PORT_OPT \
                            -U $DBRWUSER \
                             $DBNAME \
                            -c "$1"
     fi
}

```
Output 
````
root@casbc00 ~> [~]# opensipsctl dispatcher show
+ PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/bin:/usr/sbin//
+ TEST=false
+ '[' -n '' -a -f '' ']'
+ '[' -f /etc/opensips/opensipsctlrc ']'
+ . /etc/opensips/opensipsctlrc
++ DBENGINE=PGSQL
++ DBPORT=5235
++ DBHOST=dbv01.networklab.prod
++ DBNAME=opensips_prod01
++ DBRWUSER=opensips_prod
++ DBRWPW=oKxUj9WUVhSdZronoAl1tHLEDMzbGKn6
++ OSIPS_FIFO=/var/run/opensips/opensips_fifo
++ ETCDIR=/etc/opensips
+ '[' -f /usr/etc/opensips//opensipsctlrc ']'
+ '[' -f /root/.opensipsctlrc ']'
+ '[' false = true ']'
+ '[' -z ']'
+ '[' -z ']'
+ '[' -x /bin/bash ']'
+ export SHELL_TESTED=yes
+ SHELL_TESTED=yes
+ exec /bin/bash /usr/sbin/opensipsctl dispatcher show
dispatcher gateways
pgsql_query: /usr/bin/psql  -A -q -t -P fieldsep='	' -hdbv01.networklab.prod -p5235 -U opensips_prod opensips_prod01 -c 'select * FROM dispatcher ORDER BY setid; '
7       1       sip:10.18.130.26:5160              0       fs://:       0              FS casip01
6       1       sip:10.18.130.24:5160              0       fs://:       0              FS casip00

````

**Comment by volga629** [comment]
Please submit the fix

**Comment by volga629** [comment]
Also in /usr/lib64/opensips/opensipsctl/opensipsctl.pgsql need comment line 
begins with mecho make clean output.

````
40 pgsql_query() {
 41         # if password not yet queried, query it now
 42         prompt_pw "PgSQL password for user '$DBRWUSER@$DBHOST'"
 43         #mecho "pgsql_query: $PGSQL $2 -A -q -t -P fieldsep='   ' -h$DBHOST $PORT_OPT -U $DBRWUSER $DBNAME -c '$1'"
 
````

**Comment by liviuchircu** [comment]
Just pushed your fix on `master` ( abd1e655f )

**Comment by volga629** [comment]
thank you
